variables.tf

```diff
@@ -156,7 +156,7 @@ variable "private_ips" {
 }
 
 variable "public_ip" {
-  description = "OCID of the Public IP to attach to primary vnic. Valid values are NONE, RESERVED or EPHEMERAL."
+  description = "Whether to create a Public IP to attach to primary vnic and wwhich lifetime. Valid values are NONE, RESERVED or EPHEMERAL."
```

Comment:
typo. Also I think you should rename the variable to `public_ip_type`.

Reply:
I would prefer to keep it generic, so if there is a possibility later to also provide the OCID of existing reserved IP to reuse non-managed objects, it will be possible with the same input.

Reply:
ok. you need to fix the typos though